### ISSUE TITLE
Segmentation Fault from improper use of `Frame.from_csv`

### ISSUE THREAD

**Issue by Elipzer** [static-frame/static-frame#717]
**Description**

Improper specification of dtypes/index parameters in `Frame.from_csv` can cause a segmentation fault

**Example**

Provide an interactive Python session, or Python code, to demonstrate the issue. Please keep example data to a minimum.

`sf-crash.csv`
```
date_close,balance_open,balance_close
2017-06-30,0.00,193.33
2017-07-31,193.33,197.97
2017-08-31,197.97,203.02
2017-09-30,203.02,206.44
```

`sf-crash.py`
```python
from decimal import Decimal

import static_frame as sf

sf.Frame.from_csv(
    "./sf-crash.csv",
    dtypes=["datetime64[D]", "datetime64[D]", str, str],
    index_depth=2,
    index_constructors=[sf.IndexDate, sf.IndexDate],
).assign[["balance_open", "balance_close"]].apply_element(Decimal)
```

```
$ python sf-crash.py
zsh: segmentation fault (core dumped)  python sf-crash.py
```

**Platform**

Run the following function (static-frame >= 0.8.1) and provide the results to define your platform and environment:

```python
>>> import static_frame as sf
>>> sf.Platform.display()
<Series: platform>
<Index>
platform           Linux-6.1.6-arch1-1-x86_64-with-glibc2.36
sys.version        3.10.9 (main, Dec 19 2022, 17:35:49) [GCC 12.2.0]
static-frame       1.0.1
numpy              1.23.5
pandas             1.5.3
xlsxwriter         <ModuleNotFoundError>
openpyxl           <ModuleNotFoundError>
xarray             <ModuleNotFoundError>
tables             <ModuleNotFoundError>
pyarrow            <ModuleNotFoundError>
msgpack            (1, 0, 3)
msgpack_numpy      <ModuleNotFoundError>
<<U13>             <object>
```

**Comment by flexatone** [comment]
Many thanks for reporting this issue. I will investigate further as a segmentation fault should never occur in this scenario.

**Comment by flexatone** [comment]
Thanks again for posting this issue.

This appear to be an issue with NumPy, or the version of NumPy to which the `arraykit` package is linked. 

With NumPy 1.19.5, I get the following in StaticFrame, running the posted example (which is expected):
```
ValueError: Error parsing datetime string "203.02" at position 3
```

With both NumPy 1.23.5 and 1.24.1 I can reproduce the segmentation fault when I run StaticFrame, but cannot repeat the error in an interactive session like this:

```
>>> np.datetime64('203.02')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
ValueError: Error parsing datetime string "203.02" at position 3
```

This might show that it is an issue with how `arraykit` package links to NumPy. I will explore some options there.

**Comment by flexatone** [comment]
See also:
https://github.com/static-frame/arraykit/pull/88

**Comment by flexatone** [comment]
Unfortunately, the build of `arraykit` did not solve this issue. I will look into seeing of there is anything else that can be done in `arraykit` to avoid this.

**Comment by flexatone** [comment]
This issue has been resolved with the new  version of ArrayKit 0.2.9, which will be updated upcoming 1.2.3 StaticFrame release.
